source/extensions/filters/http/common/aws/credentials_provider.h

```diff
@@ -13,17 +13,27 @@ namespace HttpFilters {
 namespace Common {
 namespace Aws {
 
+/**
+ * AWS credentials container
+ *
+ * If a credential component was not found in the execution environment, it's getter method will
+ * return absl::nullopt. Credential components with the empty string value are treated as not found.
+ */
 class Credentials {
```

Comment:
I felt like this should be a struct but not feeling strongly about it.

Reply:
Interface of this type is not public by default. Data members specifically are not public and can not be changed once set by the constructor. In that sense it feels more like a type (class) than a collection of data (struct).